### PR TITLE
Add external-link mode for target=blank

### LINKED ISSUE #3480: Potential security vulnerability: target="_blank" ?
hi,

I just came accross this:
https://developers.google.com/web/tools/lighthouse/audits/noopener

and realized that external links simply have target="_blank" in the output

what's your take on the situation?

### PR DESCRIPTION
**Pull request by jelovirt**
Add `external-link` mode to generate `target="blank"` attribute in HTML5. This allows easy extension to include e.g. `rel="noopener"` attribute to external links.

Fixes #3480

### CODE REVIEW

**Comment by infotexture** [comment]
While we're at it, should we also add the `external` keyword to the `@rel` value ? 🤔 

```html
<a class="xref" href="https://brew.sh" rel="external noopener" target="_blank">Homebrew</a>
```

This would help to preserve the semantic information from the source file and support dedicated styling for external links with icons to indicate that the link leads somewhere else:

```css
a[rel="external"] {
  background-image: url("…")
}
```

File: src/main/plugins/org.dita.html5/xsl/topic.xsl [hunk 1]
```diff
@@ -590,6 +590,9 @@ See the accompanying LICENSE file for applicable license.
     </blockquote>
   </xsl:template>
   
+  <xsl:template match="node()" mode="external-link" as="attribute()*">
+    <xsl:attribute name="target">_blank</xsl:attribute>
```
[the remark this answers is not in the document]
Comment (robander) [reply src/main/plugins/org.dita.html5/xsl/topic.xsl]: That sounds fine to me - I think it's a good idea to do it as you've shown, keeping this common template for external-role attributes, just add that one by default.

**Comment by thendarion** [comment]
> While we're at it, should we also add the `external` keyword to the `@rel` value ? 🤔

if you want to do something like that,  then ```class="xref external"``` is more natural

**Comment by infotexture** [comment]
@thendarion We should defer to the HTML standard, which explicitly defines a series of [link types](https://html.spec.whatwg.org/multipage/links.html#linkTypes):

> The types of link indicated (the relationships) are given by the value of the `rel` attribute

The [external](https://html.spec.whatwg.org/multipage/links.html#link-type-external) keyword is designed for this very purpose, so `rel="external"` is preferable.

**Comment by jelovirt** [comment]
GitHub's not showing the [CI status](https://travis-ci.org/github/dita-ot/dita-ot/builds/665916778)

**Pull request merged**